Framework/jmaqs-appium/src/test/java/com/magenic/jmaqs/appium/AppiumUtilitiesUnitTest.java

```diff
@@ -0,0 +1,215 @@
+
+    // Assert screenshot was successful
+    Assert.assertTrue(isSuccess, "Expected Screenshot to be successful.");
+  }
+
+  @Test
+  public void testSavePageSourceCustomDirectoryFileName() {
+    AppiumDriver<WebElement> appiumDriver = AppiumDriverFactory.getDefaultMobileDriver();
+    FileLogger fileLogger = (FileLogger)this.getTestObject().getLog();
+    AppiumTestObject testObject = new AppiumTestObject(
+        appiumDriver,
+        fileLogger,
+        this.getTestObject().getFullyQualifiedTestName());
+    this.setTestObject(testObject);
+
+    // Open Google and take a screenshot with a custom name and date/time appended
+    appiumDriver.navigate().to("http://www.google.com");
+    String dateTime = DateTimeFormatter.ofPattern(
+        "uuuu-MM-dd-HH-mm-ss-SSSS",
+        Locale.getDefault()).format(LocalDateTime.now(Clock.systemUTC()));
+    String filePath = AppiumUtilities.savePageSource(
+        appiumDriver,
+        testObject,
+        fileLogger.getDirectory(),
+        StringProcessor.safeFormatter("%s - %s","TestCustomName", dateTime));
+
+    // Assert File Path returned from Screenshot is the same as expected file path.
+    Assert.assertEquals(
+        filePath,
+        Paths.get(StringProcessor.safeFormatter(
+            "%s%s - %s%s",fileLogger.getDirectory(), "\\TestCustomName", dateTime, ".txt")).normalize().toString());
+  }
+
+  @Test
+  public void testSavePageSourceConsoleLogger() {
+    AppiumDriver<WebElement> appiumDriver = AppiumDriverFactory.getDefaultMobileDriver();
+    ConsoleLogger consoleLogger = new ConsoleLogger();
+    AppiumTestObject testObject = new AppiumTestObject(
+        appiumDriver,
+        consoleLogger,
+        this.getTestObject().getFullyQualifiedTestName());
+    this.setTestObject(testObject);
+
+    // Open Google and take a screenshot
+    appiumDriver.navigate().to("http://www.google.com");
+    boolean isSuccess = AppiumUtilities.savePageSource(appiumDriver, testObject, "testAppend");
+
+    // Assert screenshot was successful
+    Assert.assertTrue(isSuccess, "Expected Screenshot to be successful.");
+  }
+
+  @Test
+  public void testKillDriver() {
+    AppiumDriver<WebElement> appiumDriver = AppiumDriverFactory.getDefaultMobileDriver();
+    FileLogger fileLogger = (FileLogger)this.getTestObject().getLog();
+    AppiumTestObject testObject = new AppiumTestObject(
+        appiumDriver,
+        fileLogger,
+        this.getTestObject().getFullyQualifiedTestName());
+    this.setTestObject(testObject);
+
+    // Open Google then Kill the Driver
+    appiumDriver.navigate().to("http://www.google.com");
+    AppiumUtilities.killDriver(appiumDriver);
+
+    // Assert that the Session ID is null in the Appium Driver
+    Assert.assertNull(appiumDriver.getSessionId(), "Expected Appium Driver Session ID to be null.");
+  }
+
+  @Override protected void postSetupLogging() throws Exception {
```

Comment:
Feel free to remove empty methods.  We most likely will not use them.

Reply:
I am extending BaseTest, which makes me have to implement the abstract classes. They are just empty since there is no need to have anything in it. Should I extend a different class instead?

Reply:
Nah.  Stick with that for now.  We'll be having deprecation clean up on the back end of December which will get rid of `postSetupLogging()`.  We'll figure out what to do about teardown method soonish.